src/videoCache.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * This module interacts with the server used to cache video ad content to be restored later.
+ * At a high level, the expected workflow goes like this:
+ *
+ *   - Request video ads from Bidders
+ *   - Generate IDs for each valid bid, and cache the key/value pair on the server.
+ *   - Return these IDs so that publishers can use them to fetch the bids later.
+ *
+ * This trickery helps integrate with ad servers, which set character limits on request params.
+ */
+
+import { ajax } from './ajax';
+import { tryAppendQueryString } from './utils';
+
+const PUT_URL = 'https://prebid.adnxs.com/pbc/v1/put'
+const GET_URL = 'https://prebid.adnxs.com/pbc/v1/get?';
+const EMPTY_VAST_RESPONSE = '<VAST version="3.0"></VAST>'
+
+/**
+ * These are the properties required on a Bid in order to cache and retrieve it.
+ *
+ * @typedef {object} CacheableBid
+ * @property {string} vastUrl A URL which loads some valid VAST XML.
+ */
+
+/**
+ * Function which wraps a URI that serves VAST XML, so that it can be loaded.
+ *
+ * @param {string} uri The URI where the VAST content can be found.
+ * @return A VAST URL which loads XML from the given URI.
+ */
+function wrapURI(uri) {
+  // Technically, this is vulnerable to cross-script injection by sketchy vastUrl bids.
+  // We could make sure it's a valid URI... but since we're loading VAST XML from the
+  // URL they provide anyway, that's probably not a big deal.
+  return '<VAST version="2.0"><Ad id=""><Wrapper><AdSystem>prebid.org wrapper</AdSystem><VASTAdTagURI><![CDATA[' +
+    uri +
+    ']]></VASTAdTagURI><Impression></Impression><Creatives></Creatives></Wrapper></Ad></VAST>'
+}
+
+/**
+ * Wraps a bid in the format expected by the prebid-server endpoints, or returns null if
+ * the bid can't be converted cleanly.
+ *
+ * @param {CacheableBid} bid
+ */
+function toStorageRequest(bid) {
+  return {
+    'value': wrapURI(bid.vastUrl)
+  };
+}
+
+/**
+ * If the server responded successfully, then it gave us a bunch of objects with uuid properties.
+ * Abstract our background dependency away by renaming them to cacheId.
```

Comment:
I ended up removing this. In part because you found the comment confusing... also in part because I took a step back and realized that only the `bidmanager` sees it anyway. Not a part of the public API.